File: SL-lab/Lab8/test/prg2.py
def betterSurround(st):
    st = list(st)
    
    if st[0]!= '+' and st[len(st)-1]!= '=':
        return False
    else:
        for i in range(0,len(st)):
            if st[i].isalpha():
                if not (st[i-1]=='+' or st[i-1]=='=') and (st[i+1] == '+' or st[i+1] == '='):
                    return false
            else:
                if i!=0 and i!= len(st)-1:
                    if st[i-1].isalpha() and st[i+1].isalpha():
                        return False
    return True

new = betterSurround("+f=+d+")
if new:
    print("y")
else:
    print("N")

File: SL-lab/Lab8/test/prg2.js
function betterSurround(str)
{
    if((str.charAt(0) != '=' && str.charAt(0) != '+') || (str.charAt(str.length-1) != '=' && str.charAt(str.length-1) != '+'))
        return false
    else
    {
        for(var i =1;i<str.length-1;i++)
        {
            if(str.charAt(i)>='a' && str.charAt(i)<='z')
            {
                if(!(((str.charAt(i-1) == '+' || str.charAt(i-1) == '=') && (str.charAt(i+1) == '+' || str.charAt(i+1) == '='))))
                    return false
            }    
            else if(str.charAt(i-1)>='a' && str.charAt(i-1)<='z' && str.charAt(i+1)>='a' && str.charAt(i+1)<='z')
                return false

        }
    }
    return true
}

var newwrd = betterSurround("+f+=d+")

if(newwrd)
    console.log("y")
else
    console.log("n")
